include/ideep/operators/inner_product.hpp

```diff
@@ -560,17 +558,6 @@ struct inner_product_backward_data : public dnnl::inner_product_backward_data {
                       const attr_t& attr = attr_t(),
                       const engine& aengine = engine::cpu_engine()) {
     auto weights_ = weights;
-    if (diff_dst.get_data_type() == data_type::bf16) {
-      if (weights_.get_data_type() != data_type::bf16) {
-        weights_.init(weights.get_desc().to_type(data_type::bf16));
-        weights_.reorder_from(weights);
-      }
-    } else if (diff_dst.get_data_type() == data_type::f16) {
-      if (weights_.get_data_type() != data_type::f16) {
-        weights_.init(weights.get_desc().to_type(data_type::f16));
-        weights_.reorder_from(weights);
-      }
-    }
```

Comment:
These lines can be safely deleted?

Reply:
> These lines can be safely deleted?

I think so. The purpose of these lines is to get a weight tensor in `f16` data type and use its desc for pd creation. But actually, we don't need to do reorder here. We can simply create a weight desc by shape, data type, and `tag::any` for pd creation (It's a suggestion from onednn team when we were discussing the `to_type` issue).
The weight tensor will then be compared with expected desc (queried from pd) and reorder if needed.